### DISCUSSION TITLE
Restart useAsync if query was changed

### DISCUSSION THREAD

**Discussion by JuliWolf** [nuxt/bridge#459] [General]
Hello.

I'm trying to migrate my app to vue3 syntax using @nuxtjs/composition-api

How I can restart useAsync method like it was in nuxt 2 with asyncData using watchQuery

**Comment by antlionguard** [comment]
Like this: 


```ts
const route = useRoute();
const fullPath = computed(() => route.fullPath); 
useLazyAsyncData(
   'key',
   'fetchMethodHere',
   {
     watch: [fullPath]
   }
);
```

**Reply by antlionguard** [reply]
And you can pass through ref values inside `watch` array.
